packages/map-template/src/components/Map/Map.jsx

```diff
@@ -80,16 +87,32 @@ function Map({ apiKey, gmApiKey, mapboxAccessToken, venues, venueName, onLocatio
         return mapsIndoorsInstance.fitVenue(venue);
     }
 
+    /**
+     * When the building changes, replace the default tile URL style to the incoming tile style.
+     * Set the MapsIndoors Tile URL to the incoming tile style.
+     */
+    const onBuildingChanged = (miInstance) => {
+        if (miInstance) {
+            const tileURL = miInstance.getTileURL().replace('lundefined/', 'l{floor}/').replace(/\/l(\d+)\//gm, '/l{floor}/');
```

Comment:
Maybe you can try to formulate in a comment why the replacement of "lundefined/" is necessary 😄 Try to experiment with what happens if we omit it to get a better understanding.

Reply:
Simplified the function and removed the unnecessary replacements.